src/NodeCompiler/CompileNodeToValue.php

```diff
@@ -60,7 +60,7 @@ public function __invoke(Node $node, CompilerContext $context)
     /**
      * Compile constant expressions
      *
-     * @return bool|mixed|null
+     * @return scalar|scalar[]|null
```

Comment:
I think we should use `@psalm-` -prefixed annotations for psalm-specific types, to avoid confusing existing tools like @phpstan or @jetbrains